### ISSUE TITLE
Create wrapper for 1 or more standard crypto currency APIs

### ISSUE THREAD

**Issue by msgilligan** [JavaMoney/javamoney-shelter#19]
If it's possible to create a JavaMoney SPI-compatible wrapper for a standard crypto currency API like [Xchange](https://github.com/timmolter/XChange), JavaMoney could quickly support a large number of exchanges.

I have not studied the feasibility of this at all, but wanted to propose the idea as an issue so others could comment.

**Comment by keilw** [comment]
Sounds like a good idea. While Xchange is built against Java 6, especially the Backports (the API should in theory work as far back as Java 6, I think there was no "Diamond" usage either) should work fine.

**Comment by keilw** [comment]
Yes, the API-BP has  <jdkVersion>1.6</jdkVersion> so only in the BP of the RI there are references to Java 7. Creating e.g. an independent Android port should therefore work. And in real life most likely a lot of apps uses Java 7 anyway.

**Comment by msgilligan** [comment]
I've implemented a basic version of this over in http://github.com/msgilligan/bitcoinj-addons.

`BaseXChangeExchangeRateProvider.java` does the heavy lifting: https://github.com/msgilligan/bitcoinj-addons/blob/master/bitcoinj-money/src/main/java/com/msgilligan/bitcoinj/money/BaseXChangeExchangeRateProvider.java

I started with wrapper classes for each exchange built on top of this, but I recently added `DynamicXChangeRateProvider.java` which can wrap any XChange provider specified by classname string:   https://github.com/msgilligan/bitcoinj-addons/blob/master/bitcoinj-money/src/main/java/com/msgilligan/bitcoinj/money/DynamicXChangeRateProvider.java

**Comment by msgilligan** [comment]
I've also very-softly-launched [OmniPortfolio](http://www.omniportfolio.org) which uses this stuff. The UI and the site need more work before I'll even consider it soft-launched, but there is a public-facing text-only website now.
